packages/vscode-extension/lib/runtime.js

```diff
@@ -25,15 +25,74 @@ global.__RNIDE_onDebuggerReady = function () {
 // debug adapter and avoid exposing as part of application logs
 console.log("__RNIDE_INTERNAL", "radon-ide runtime loaded");
 
+let stackOffset = 1; // default offset is 1, because the first frame is the wrapConsole function
+let logRef = null;
+let errorRef = null;
+let warnRef = null;
+let infoRef = null;
+
+function isConsoleRefChanged() {
+  return (
+    logRef !== console.log ||
+    errorRef !== console.error ||
+    warnRef !== console.warn ||
+    infoRef !== console.info
+  );
+}
+
+function updateConsoleRefs() {
+  logRef = console.log;
+  warnRef = console.warn;
+  errorRef = console.error;
+  infoRef = console.info;
+}
```

Comment:
why can't we just use it for each function separately? This could just be defined inside wrapConsole and only work for the specific console function? Otherwise we're assuming that all log types are wrapped the same way which doesn't have to be the case.

In my optinion it'd be simpler to have it run for each function individually and would also not require this assumption.